chia/wallet/wallet_state_manager.py

```diff
@@ -1034,12 +1034,10 @@ async def new_coin_state(
         used_up_to = -1
         ph_to_index_cache: LRUCache = LRUCache(100)
 
-        local_records: List[Optional[WalletCoinRecord]] = await self.coin_store.get_coin_records(
-            [st.coin.name() for st in coin_states]
-        )
+        local_records = await self.coin_store.get_coin_records([st.coin.name() for st in coin_states])
 
-        assert len(local_records) == len(coin_states)
-        for coin_state, local_record in zip(coin_states, local_records):
+        for coin_state in coin_states:
+            local_record = local_records.get(coin_state.coin.name())
```

Comment:
I notice we call `Coin.name()` 4 times per coin state now. we used to call it 3 times. it's not trivial to compute.